File: add_padding.py
from PIL import Image
import os

def add_padding(input_path, output_path, padding_ratio=0.3, background_color=(0, 0, 0)):
    try:
        img = Image.open(input_path)
        original_width, original_height = img.size
        
        # Calculate new size (keeping aspect ratio)
        # We want the original image to be (1 - padding_ratio) of the canvas
        # So if padding_ratio is 0.3 (30%), the image will be 70% of the canvas
        scale_factor = 1.0 - padding_ratio
        new_width = int(original_width * scale_factor)
        new_height = int(original_height * scale_factor)
        
        # Resize the image
        img_resized = img.resize((new_width, new_height), Image.Resampling.LANCZOS)
        
        # Create new background
        background = Image.new('RGB', (original_width, original_height), background_color)
        
        # Calculate position to center
        x_offset = (original_width - new_width) // 2
        y_offset = (original_height - new_height) // 2
        
        # Paste resized image onto background
        background.paste(img_resized, (x_offset, y_offset))
        
        # Save
        background.save(output_path, quality=95)
        print(f"Successfully created padded icon at {output_path}")
        
    except Exception as e:
        print(f"Error processing image: {e}")

if __name__ == "__main__":
    input_file = "assets/icon.jpeg"
    output_file = "assets/icon_padded.jpeg"
    
    if os.path.exists(input_file):
        add_padding(input_file, output_file)
    else:
        print(f"Input file not found: {input_file}")

File: bin/add_padding.dart
import 'dart:io';
import 'package:image/image.dart';

void main() async {
  const inputPath = 'assets/images/logo.jpeg';
  const outputPath = 'assets/images/logo_padded.jpeg';
  const paddingRatio = 0.4; // 40% padding

  final inputFile = File(inputPath);
  if (!inputFile.existsSync()) {
    print('Input file not found: $inputPath');
    exit(1);
  }

  print('Reading image from $inputPath...');
  final image = decodeImage(inputFile.readAsBytesSync());

  if (image == null) {
    print('Failed to decode image');
    exit(1);
  }

  final originalWidth = image.width;
  final originalHeight = image.height;

  // Calculate new size (keeping aspect ratio)
  // We want the original image to be (1 - paddingRatio) of the canvas
  final scaleFactor = 1.0 - paddingRatio;
  final newWidth = (originalWidth * scaleFactor).round();
  final newHeight = (originalHeight * scaleFactor).round();

  print('Resizing image to ${newWidth}x$newHeight...');
  final resizedImage = copyResize(
    image,
    width: newWidth,
    height: newHeight,
    interpolation: Interpolation.cubic,
  );

  // Create new background (black)
  final background = Image(
    width: originalWidth,
    height: originalHeight,
    numChannels: 3, // RGB
  );
  // Fill with #fafafa (250, 250, 250)
  fill(background, color: ColorRgb8(250, 250, 250));

  // Calculate position to center
  final xOffset = (originalWidth - newWidth) ~/ 2;
  final yOffset = (originalHeight - newHeight) ~/ 2;

  // Paste resized image onto background
  print('Compositing image...');
  compositeImage(
    background,
    resizedImage,
    dstX: xOffset,
    dstY: yOffset,
  );

  // Save
  print('Saving to $outputPath...');
  File(outputPath).writeAsBytesSync(encodeJpg(background, quality: 95));
  print('Successfully created padded icon at $outputPath');
}
